### ISSUE TITLE
Updating the comparison with Tokio's new (2019-10-13) scheduler

### ISSUE THREAD

**Issue by Qqwy** [Xudong-Huang/may#72]
Tokio is switching to a new scheduler implementation. Read ["Making the Tokio Scheduler 10x faster"](https://tokio.rs/blog/2019-10-scheduler/) for the details.

As such, it would be nice if a new comparison/benchmark of between May and Tokio could take place, as it means that the 'Performance' section of the current README will become outdated.

**Comment by Xudong-Huang** [comment]
Great article! Sure, I will update the doc when new tokio settle down.

**Comment by zhuxiujia** [comment]
> Tokio is switching to a new scheduler implementation. Read ["Making the Tokio Scheduler 10x faster"](https://tokio.rs/blog/2019-10-scheduler/) for the details.
> 
> As such, it would be nice if a new comparison/benchmark of between May and Tokio could take place, as it means that the 'Performance' section of the current README will become outdated.

cant beat tokio。
```rust
fn main() {
    may::config().set_workers(6).set_stack_size(0x1);
    let total=1000000;
    for _ in 0..total{
        go!(move || {
           may::coroutine::sleep(Duration::from_secs(1*60*60));
        });
    }
    println!("done");
    std::thread::sleep(Duration::from_secs(1*60*60));
}
```
this code  consume 4.44GB Memory。but in  golang，that takes  2.35GB.
but !   in  tokio,  this code takes   only  500MB.

so. i say   cant beat tokio。 Maybe it's because of  CSP mode.  High concurrence in CSP  is not high

**Comment by Xudong-Huang** [comment]
Yes, you are right, CSP consume more memory. I will update the bench data in a separate place, not in the README. For detailed io performance you can ref to https://tfb-status.techempower.com/

**Comment by Xudong-Huang** [comment]
now I place the latest simple comparison with latest tokio version under [performance.md](https://github.com/Xudong-Huang/may/blob/master/performace.md). I don't want to misleading people about real performance, people should have their own judge to select a proper tech for projects.

**Comment by zhuxiujia** [comment]
> Yes, you are right, CSP consume more memory. I will update the bench data in a separate place, not in the README. For detailed io performance you can ref to https://tfb-status.techempower.com/

I still don't understand why resource consumption is higher than golang。
```rust
may::config().set_workers(6).set_stack_size(0x1);
```
Arguably less than golang

**Comment by Xudong-Huang** [comment]
10K coroutine will consume at least 10_000 * 4K = 40M bytes memory. 1M coroutine need 4G bytes memory at least. May don't support growth stack automatically. It's very hard for a library to grow stack dynamically. Futures are stackless, that's why the rust team  choose it.
